src/main/scala/microc/util/Reflection.scala

```diff
@@ -0,0 +1,51 @@
+package microc.util
+
+import scala.reflect.runtime.universe._
+
+/*
+// USAGE
+case class Person(name: String, age: Int)
+val personFactory = new ReflectionHelpers.CaseClassFactory[Person]
+val result: Person = personFactory.buildWith(Seq("Connor", 27))
+  val expected = Person("Connor", 27)
+  assert(result == expected)
+*/
+
+
+object ReflectionHelpers {
```

Comment:
Give proper attribution for this code (taken from [this gist](https://gist.github.com/ConnorDoyle/7002426)).